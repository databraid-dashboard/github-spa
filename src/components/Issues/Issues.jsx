/* eslint-disable import/no-named-as-default */
import React, { Component } from 'react';
import { Card } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import Issue from '../Issue/Issue';
import { retrieveIssues } from '../../actions/issueActions';
import './Issues.css';

function issueComponents(issues, repo) {
  if (issues && issues[repo]) {
    return issues[repo].map(id => <Issue key={id} issueId={id} />,
    );
  }

  return '';
}

export class Issues extends Component {
  componentDidMount() {
    this.props.retrieveIssues(this.props.userName, this.props.orgName, this.props.repoName);
  }

  render() {
    if (this.props.loadingIssues) {
      return (
        <div>Loading Issues</div>
      );
    }
    return (
      <Card>
        <Card.Content>
          <Card.Header className="ui center aligned">
            Issues
          </Card.Header>
        </Card.Content>
        {issueComponents(this.props.issuesByRepo, this.props.repoName)}
      </Card>
    );
  }
}

Issues.propTypes = {
  retrieveIssues: PropTypes.func.isRequired,
  loadingIssues: PropTypes.bool,
  userName: PropTypes.string.isRequired,
  orgName: PropTypes.string.isRequired,
  repoName: PropTypes.string.isRequired,
  issuesByRepo: PropTypes.objectOf(PropTypes.array),

};

Issues.defaultProps = {
  loadingIssues: false,
  issuesByRepo: {},
};


export const mapStateToProps = state => ({
  issuesByRepo: state.issues.issuesByRepo,
  loadingIssues: state.loadingIssues,
  orgName: state.currentPage.selectedOrgName,
  userName: state.currentPage.userName,
});

export const mapDispatchToProps = dispatch => bindActionCreators({
  retrieveIssues,
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Issues);
