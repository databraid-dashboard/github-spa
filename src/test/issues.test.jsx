import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Issues, mapStateToProps, mapDispatchToProps } from '../components/Issues/Issues';


const state = {
  issues: {
    ids: [246869449],
    issuesById: {
      246869449: {
        assignedTo: [],
        labels: [],
        repoIssueNumber: 36,
        title: 'disregard this PR',

      },
    },
  },
  loadingIssues: false,
};

describe('Issues component', () => {
  it('should render a component with props as specified ', () => {
    const retrieveIssues = jest.fn();
    const component = shallow(
      <Issues
        issuesIds={state.issues.ids}
        loadingIssues={false}
        retrieveIssues={retrieveIssues}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  //
  // it('Should call the retrieveIssues Function', ()=> {
  //   const retrieveIssues = jest.fn();
  //   const store = mockStore({ })
  //   const component = mount (
  //     <Issues issuesIds={state.issues.ids}
  //       loadingIssues={false}
  //       retrieveIssues={retrieveIssues}/>,
  //   );
  //   expect(retrieveIssues).toBeCalled();
  // });

  it('map\'s given State To Props ', () => {
    const expected = {
      issuesIds: [246869449],

      loadingIssues: false,
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it('maps component dispatches to props', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('retrieveIssues');
  });
  it('Should have a Cart Header in the center', () => {
    const retrieveIssues = jest.fn();
    const component = shallow(
      <Issues
        issuesIds={state.issues.ids}
        loadingIssues={false}
        retrieveIssues={retrieveIssues}
      />,
    );
    expect(component.find('.aligned').exists()).toBe(true);
  });
});
