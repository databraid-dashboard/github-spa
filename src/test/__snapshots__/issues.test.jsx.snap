// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issues component should render a component with props as specified  1`] = `
<Card>
  <CardContent>
    <CardHeader
      className="ui center aligned"
    >
      Issues
    </CardHeader>
  </CardContent>
</Card>
`;
